server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -226,14 +226,17 @@ public Builder(String name, IndexVersion indexVersionCreated) {
                 if (v != null && dims.isConfigured() && dims.get() != null) {
                     v.validateDimension(dims.get());
                 }
-                if (v != null && v.supportsElementType(elementType.getValue()) == false) {
-                    throw new IllegalArgumentException(
-                        "[element_type] cannot be [" + elementType.getValue().toString() + "] when using index type [" + v.type + "]"
-                    );
+                if (v != null) {
+                    v.validateElementType(elementType.getValue());
                 }
             })
                 .acceptsNull()
-                .setMergeValidator((previous, current, c) -> previous == null || current == null || previous.updatableTo(current));
+                .setMergeValidator(
+                    (previous, current, c) -> previous == null
+                        || current == null
+                        || Objects.equals(previous, current)
```

Comment:
Why did we have to add the `equals` check here?

Reply:
@pmpailis because some of the update logic actually didn't truly check for full equality and we should always allow updates that are 100% equal.